dist/tools/buildsystem_sanity_check/check.sh

```diff
@@ -415,7 +415,7 @@ all_checks() {
 
 main() {
     all_checks | prepend 'Invalid build system patterns found by '"${0}:" | error_on_input >&2
-    exit $?
+    return $?
```

Comment:
Ah, this is a functional change. Now the calls to `github_annotate_teardown` and `github_annotate_report_last_run` below become reachable.